aws_lambda_powertools/utilities/data_classes/sqs_event.py

```diff
@@ -83,6 +85,8 @@ def __getitem__(self, key: str) -> Optional[SQSMessageAttribute]:  # type: ignor
 class SQSRecord(DictWrapper):
     """An Amazon SQS message"""
 
+    NestedEvent = TypeVar("NestedEvent", bound=DictWrapper)
```

Comment:
The same suggestion from [L224](https://github.com/awslabs/aws-lambda-powertools-python/pull/2349/files#diff-3724c815a859e091c2960fb04a7d8ec4fdc1d9d37b9055dc0c6541b8e0ca5424R224) applies here.